src/authenticator.rs

```diff
@@ -744,8 +756,8 @@ where
                     return Err(Status::UnspecifiedPersistentExecutionError);
                 }
             }
-            Kind::HotpReverse => {
-                // This credential kind should never be access through calculate()
+            _ => {
```

Comment:
Why switch to a placeholder rather than leaving `Kind::HotpReverse`? Also this makes clippy complain that the bellow arm is not reachable.

Reply:
I've added new kind with the same behavior, did not make sense to list them separately.
Indeed the arm below got added accidentally - it got removed in the subsequent PRs.